### PR TITLE
shake config property and ability to explicitly set the window mode

### PR DESCRIPTION
**Pull request by piemonte**
hey @Cosmo

great library! – been very helpful for some of my projects.

there's a couple minor features i personally needed which i added to my fork. figured i could send a PR incase you had interest in them too. if not, no big deal to ignore.

thanks,
patrick

### CODE REVIEW

File: TinyConsole/TinyConsoleController.swift [hunk 1]
```diff
@@ -14,9 +14,20 @@ open class TinyConsoleController: UIViewController {
...
     /// - collapsed: the console is hidden
     /// - expanded: the console is shown
-    enum WindowMode {
-        case collapsed
+    public enum WindowMode: Int, CustomStringConvertible {
+        case collapsed = 0
         case expanded
+        
+        public var description: String {
+            get {
```
Comment (mRs-) [inline TinyConsole/TinyConsoleController.swift]: remove get here, it's not necessary

File: TinyConsole/TinyConsole.swift [hunk 2]
```diff
@@ -26,10 +30,26 @@ open class TinyConsole {
         return formatter
     }()
     
+    var shakeEnabled: Bool {
+        get {
+            return TinyConsole.shared.tinyConsoleController!.shakeEnabled
```
Comment (mRs-) [inline TinyConsole/TinyConsole.swift]: do not use force casts!

File: TinyConsole/TinyConsoleViewController.swift [hunk 3]
```diff
@@ -22,6 +22,10 @@ class TinyConsoleViewController: UIViewController {
         TinyConsole.shared.textView = consoleTextView
         view.addSubview(consoleTextView)
         
+        if let parent = self.parent {
+            TinyConsole.shared.tinyConsoleController = parent as? TinyConsoleController
+        }
```
Comment (mRs-) [inline TinyConsole/TinyConsoleViewController.swift]: you can drop the if let cast here, it's not necessary

File: TinyConsole/TinyConsole.swift [hunk 4]
```diff
@@ -26,10 +30,26 @@ open class TinyConsole {
...
+    }
+    
     func currentTimeStamp() -> String {
         return dateFormatter.string(from: Date())
     }
     
+    public static func update(consoleMode mode: TinyConsoleController.WindowMode) {
+        if let viewController = shared.tinyConsoleController {
+            viewController.update(windowMode: mode, animated: true)
+        }
```
Comment (mRs-) [inline TinyConsole/TinyConsole.swift]: if let cast is not needed here, just chain it via optionals

File: TinyConsole/TinyConsoleController.swift [hunk 5]
```diff
@@ -14,9 +14,20 @@ open class TinyConsoleController: UIViewController {
     ///
     /// - collapsed: the console is hidden
     /// - expanded: the console is shown
-    enum WindowMode {
-        case collapsed
+    public enum WindowMode: Int, CustomStringConvertible {
+        case collapsed = 0
         case expanded
+        
+        public var description: String {
```
Comment (mRs-) [inline TinyConsole/TinyConsoleController.swift]: is it only for debugging or why exactly is this here? there is not even one call to description.

**Comment by piemonte** [comment]
@mRs- thanks for taking the time to review. pushed the requested changes, good comments. 👍

**Comment by mRs-** [comment]
Sorry for the long waiting time, I will look into this ASAP. Currently @Cosmo doesn't have time for this so I am taking over :)

File: TinyConsole/TinyConsole.swift [hunk 6]
```diff
@@ -32,8 +32,10 @@ open class TinyConsole {
     
     var shakeEnabled: Bool {
         get {
-            return TinyConsole.shared.tinyConsoleController!.shakeEnabled
```
Comment (ohitsdaniel) [inline TinyConsole/TinyConsole.swift]: return TinyConsole.shared.tinyConsoleController?.shakeEnabled ?? false

**Comment by ohitsdaniel** [comment]
I don't really see a reason why WindowMode should be exposed. Can you name any? 

Also, see my comment on commit bfdea83, using the ?? operator makes your code more concise and it does the same thing.

File: TinyConsole/TinyConsole.swift (hunk 6 above)
Comment (piemonte) [inline TinyConsole/TinyConsole.swift]: thanks @ohitsdaniel, nice comment – pushed the change

**Comment by piemonte** [comment]
hey @ohitsdaniel i appreciate your time. i originally had implemented a "toggle" func but i need to explicitly put the window mode into a specific state under various circumstances.

all of these are minor changes i've done for my use in various projects, if none of this is useful to contribute back that's fine too.

**Pull request closed without merging**